src/osd/PrimaryLogPG.cc

```diff
@@ -2344,6 +2344,14 @@ void PrimaryLogPG::do_op(OpRequestRef& op)
   maybe_force_recovery();
 }
 
+bool PrimaryLogPG::is_dedup_chunk(const chunk_info_t& chunk)
+{
+  if (pool.info.get_fingerprint_type() != pg_pool_t::TYPE_FINGERPRINT_NONE) {
+    return true;
+  }
+  return false;
```

Comment:
The only way this can happen is if pool.get_fingerprint_type() is none in SET_CHUNK, or the user changes the pool fingerprint type.  I guess I just really don't understand what it means to have the FLAG_HAS_REFERENCE flag but not the FLAG_HAS_FINGERPRINT flag.

This goes back to my previous question: do you intend for chunk_map to be able to point at things other than an immutable object with a name resulting from applying the pool fingerprint algorithm to the corresponding extent?  Is it valid for a user of SET_CHUNK to set a chunk with some arbitrary name?  Is chunk_map supposed to be able to point at something that is mutable, or isn't refcounted at all?  I just don't know what you are going for here. 

If chunk_map only contains references to refcounted immutable dedup chunks, then this should always return true, and should therefore just be omitted in its only caller.